src/main/java/com/compilercharisma/chameleonbusinessstudio/repository/UserRepository.java

```diff
@@ -112,5 +112,16 @@ public Mono<UserAppointments> getUserAppointments(String _id) {
         return vendiaClient.executeQuery(getUserAppointmentArray, "get_User", UserAppointments.class);
     }
 
+    /**
+     * Update a user's appointments after being booking an appointment
+     * @param appointments String that contains the appointments
+     * @return {@link Mono} of a {@link User}
+     */
+    public Mono<User> updateAppointmentsForUser(String userId, String appointments) {
+        var query = "mutation { update_User(id: \"%s\", input: {appointments: %s}) { result { _id displayName email phoneNumber role appointments } } }"
```

Comment:
Are we using the user's appointment array anywhere?

Reply:
Yep, the user's appointment array should be populated with the given appointment id once they are booked for one appointment, either by themselves or by an Organizer/Admin/Talen and what not, previously it was not being updated.